src/System.Linq/tests/MinMaxTests.cs

```diff
@@ -0,0 +1,373 @@
+            Assert.Equal(-10, minusTen.Min());
+            Assert.Equal(100, hundred.Min());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<int>().Min());
+        }
+
+        [Fact]
+        public void MaxInt()
+        {
+            var ten = Enumerable.Range(1, 10).ToArray();
+            var minusTen = new[] { -100, -15, -50, -10 };
+            var thousand = new[] { -16, 0, 50, 100, 1000 };
+            Assert.Equal(10, ten.Max());
+            Assert.Equal(-10, minusTen.Max());
+            Assert.Equal(1000, thousand.Max());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<int>().Max());
+        }
+
+        [Fact]
+        public void MinLong()
+        {
+            var one = Enumerable.Range(1, 10).Select(i => (long)i).ToArray();
+            var minusTen = new[] { -1L, -10, 10, 200, 1000 };
+            var hundred = new[] { 3000L, 100, 200, 1000 };
+            Assert.Equal(1, one.Min());
+            Assert.Equal(-10, minusTen.Min());
+            Assert.Equal(100, hundred.Min());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<long>().Min());
+        }
+
+        [Fact]
+        public void MaxLong()
+        {
+            var ten = Enumerable.Range(1, 10).Select(i => (long)i).ToArray();
+            var minusTen = new[] { -100L, -15, -50, -10 };
+            var thousand = new[] { -16L, 0, 50, 100, 1000 };
+            Assert.Equal(10, ten.Max());
+            Assert.Equal(-10, minusTen.Max());
+            Assert.Equal(1000, thousand.Max());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<long>().Max());
+        }
+
+        [Fact]
+        public void MinSingle()
+        {
+            var one = Enumerable.Range(1, 10).Select(i => (float)i).ToArray();
+            var minusTen = new[] { -1F, -10, 10, 200, 1000 };
+            var hundred = new[] { 3000F, 100, 200, 1000 };
+            Assert.Equal(1L, one.Min());
+            Assert.Equal(-10L, minusTen.Min());
+            Assert.Equal(100L, hundred.Min());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<float>().Min());
+        }
+
+        [Fact]
+        public void MaxSingle()
+        {
+            var ten = Enumerable.Range(1, 10).Select(i => (float)i).ToArray();
+            var minusTen = new[] { -100F, -15, -50, -10 };
+            var thousand = new[] { -16F, 0, 50, 100, 1000 };
+            Assert.Equal(10F, ten.Max());
+            Assert.Equal(-10F, minusTen.Max());
+            Assert.Equal(1000F, thousand.Max());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<float>().Max());
+        }
+
+        [Fact]
+        public void MinDouble()
+        {
+            var one = Enumerable.Range(1, 10).Select(i => (double)i).ToArray();
+            var minusTen = new[] { -1D, -10, 10, 200, 1000 };
+            var hundred = new[] { 3000D, 100, 200, 1000 };
```

Comment:
The 'D' here on the first element is pretty sublte and easy to miss or even accidentally remove, at which point the test changes.  How about explicitly typing the array, i.e. ```new double[]``` instead of ```new []```?